### PR TITLE
Handle errors in the worker processes in the graph construction pipeline

### PR DESCRIPTION
**Pull request by zheng-da**
*Description of changes:*
If a worker process fails, the entire graph construction pipeline will hang. If a worker fails, we should fail the entire graph construction pipeline immediately.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/utils.py [hunk 1]
```diff
@@ -182,6 +183,9 @@ def worker_fn(worker_id, task_queue, res_queue, user_parser):
             gc.collect()
     except queue.Empty:
         pass
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        print(e, flush=True)
```
Comment (classicsong) [inline python/graphstorm/gconstruct/utils.py:187]: use logging.error?

File: python/graphstorm/gconstruct/utils.py [hunk 2]
```diff
@@ -246,6 +250,8 @@ def multiprocessing_data_read(in_files, num_processes, user_parser):
         return_dict = {}
         while len(return_dict) < num_files:
             file_idx, vals= res_queue.get()
+            if vals is None:
+                raise RuntimeError("One of the worker processes fails. Stop data processing.")
```
Comment (classicsong) [inline python/graphstorm/gconstruct/utils.py:254]: It seems file_idx set in L188 is not used.
BTW, why not raise the RuntimeError in L188?

Comment (zheng-da) [reply python/graphstorm/gconstruct/utils.py:254]: the main idea is that we need to capture all exceptions in the worker processes and stop the master process whenever an error happens in the worker process.

**Review by classicsong — APPROVED** [review]
LGTM

**Pull request merged**